crates/jstz_node/src/services/logs/mod.rs

```diff
@@ -1,85 +1,227 @@
-use crate::tailed_file::TailedFile;
+use crate::{tailed_file::TailedFile, Error, Result};
 use actix_web::{
     get,
-    web::{Data, Path, ServiceConfig},
-    Responder, Scope,
+    web::{Data, Path, Query, ServiceConfig},
+    HttpResponse, Responder, Scope,
 };
-use jstz_crypto::public_key_hash::PublicKeyHash;
-use jstz_proto::js_logger::{LogRecord, LOG_PREFIX};
 
-use std::io::{Error, ErrorKind::InvalidInput, Result};
+use anyhow;
+use jstz_proto::context::account::Address;
+use jstz_proto::{
+    js_logger::{LogRecord, LOG_PREFIX},
+    request_logger::{RequestEvent, REQUEST_END_PREFIX, REQUEST_START_PREFIX},
+};
+use serde::{Deserialize, Serialize};
+use std::io::{self, ErrorKind::InvalidInput};
 use std::sync::Arc;
 use tokio::task::JoinHandle;
 use tokio_util::sync::CancellationToken;
 
-use self::broadcaster::Broadcaster;
-
 use super::Service;
 
 pub mod broadcaster;
+mod db;
+
+use self::{broadcaster::Broadcaster, db::Db};
+
+const DEAULT_PAGINATION_LIMIT: usize = 100;
+const DEAULT_PAGINATION_OFFSET: usize = 0;
 
 #[get("{address}/stream")]
 async fn stream_logs(
     broadcaster: Data<Broadcaster>,
     path: Path<String>,
-) -> Result<impl Responder> {
+) -> io::Result<impl Responder> {
     let address = path.into_inner();
 
-    // validate address
     let address =
-        PublicKeyHash::from_base58(&address).map_err(|e| Error::new(InvalidInput, e))?;
+        Address::from_base58(&address).map_err(|e| io::Error::new(InvalidInput, e))?;
```

Comment:
Add a TODO here for supporting better errors from the node (notably validation errors)